### PR TITLE
Add DJ-MCP server

### PR DESCRIPTION
**Pull request by cmgzn**
Provides two flexible approaches to expose Data-Juicer functionality.

Future Enhancements:

More comprehensive configuration options (e.g., export_path, dataset).
Multi-threading/processing support.
Further refinement of ops_modality_mapping.json.

### CODE REVIEW

**Review by yxdyc — COMMENTED** [review]
The docs related to this mcp server can be added into [dj-service](https://modelscope.github.io/data-juicer/en/main/docs/DJ_service.html) later.

**Comment by cmgzn** [comment]
我更新了 MCP 服务器相关的使用描述，并为两种服务器分别起了名字：Recipe-Flow 和 Granular-Operators。我不知道这两个名字是否合适，希望大家可以给出看法。

目前，DJ-MCP 服务可以通过 Python 解释器启用。我认为这对从源码安装 DataJuicer 的用户来说是一个优势，因为他们可以灵活地指定 MCP 服务运行的环境，从而更好地适应各类数据处理过程中对依赖环境的不同需求。

然而，其他的 MCP 服务往往采用独立打包的方式发布。这种方式允许 cilent 仅通过一个 JSON 配置文件就能安装 MCP，无需提前拉取任何库代码，也不需要提前配置环境。例如：

```json
"mcpServers": {
  "git": {
    "timeout": 60,
    "command": "uvx",
    "args": [
      "mcp-server-git"
    ],
  }
}
```

DJ-MCP 也可以考虑采用类似的实现方式，也就是将这两个 DJ-MCP 独立出来分别发包。为了保证MCP中每个算子都可以成功调用，需要以完整的DJ为依赖。

考虑到 DJ 的依赖项较多，如果使用上述方式，可能会导致启动时间过长。我会继续调研并尝试这种独立打包的方案，看看是否适用于我们的项目。

File: data_juicer/core/data/dataset_builder.py [hunk 1]
```diff
@@ -228,7 +229,8 @@ def parse_cli_datapath(dataset_path) -> Tuple[List[str], List[float]]:
     # Handle empty input
     if not dataset_path or not dataset_path.strip():
         return [], []
-
+    # Convert unix style path
+    dataset_path = Path(dataset_path).as_posix()
```
Comment (cyruszhang) [inline data_juicer/core/data/dataset_builder.py]: do we have edge case that needs this conversion? let's add test case if necessary

File: tests/tools/test_mcp_server.py [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+    # export OPENAI_BASE_URL=https://dashscope.aliyuncs.com/compatible-mode/v1/
+    # export OPENAI_API_KEY=your_dashscope_key
+    api_model = 'qwen2.5-72b-instruct'
+
+    async def _run_test(self, query, target):
+
+        client = MCPClient(self.api_model)
+
+        try:
+            await client.connect_to_server('./data_juicer/tools/single_op_mcp_server.py')
```
Comment (cyruszhang) [inline tests/tools/test_mcp_server.py]: naming error? no single_op_mcp_server.py is defined

File: tests/tools/test_mcp_server.py [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+
+class MCPServerTest(DataJuicerTestCaseBase):
+    # before running this test, set below environment variables:
+    # export OPENAI_BASE_URL=https://dashscope.aliyuncs.com/compatible-mode/v1/
+    # export OPENAI_API_KEY=your_dashscope_key
+    api_model = 'qwen2.5-72b-instruct'
+
+    async def _run_test(self, query, target):
+
+        client = MCPClient(self.api_model)
```
Comment (cyruszhang) [inline tests/tools/test_mcp_server.py]: why are we mocking a new client? can we use FastMCP or mock provided by the library?

File: data_juicer/tools/dj_test.txt [hunk 4]
```diff
@@ -0,0 +1,4 @@
+text_length_filter
```
Comment (cyruszhang) [inline data_juicer/tools/dj_test.txt]: feels more like config than tools file. maybe relocate?

File: data_juicer/core/data/dataset_builder.py (hunk 1 above)
Comment (cmgzn) [reply data_juicer/core/data/dataset_builder.py]: this was initially included to prevent parsing problems with Windows paths in shlex.split. Without converting to a posix-style path, Windows paths might not be split correctly. But considering that it might affect other systems and is not related to the main feature of this PR, I have removed it. I'll look into submitting a separate PR with proper tests for this later.

File: tests/tools/test_mcp_server.py (hunk 3 above)
Comment (cmgzn) [reply tests/tools/test_mcp_server.py]: done

File: data_juicer/tools/dj_test.txt (hunk 4 above)
Comment (cmgzn) [reply data_juicer/tools/dj_test.txt]: moved to configs/mcp

File: tests/tools/test_mcp_server.py (hunk 2 above)
Comment (cmgzn) [reply tests/tools/test_mcp_server.py]: sorry, fixed

**Comment by cmgzn** [comment]
Fixed Typer dependency conflict between spaCy and MCP by pinning spaCy to 3.8.0. Still using the spaCy 3.7.0 model, but test_diversity_analysis.py is passing

**Comment by HYLcool** [comment]
Please merge the latest main branch.

File: data_juicer/utils/model_utils.py [hunk 5]
```diff
@@ -658,8 +658,7 @@ def prepare_simple_aesthetics_model(pretrained_model_name_or_path, *, return_mod
         return (model, processor)
 
 
-
-def prepare_spacy_model(lang, name_pattern="{}_core_web_md-3.7.0", **model_params):
+def prepare_spacy_model(lang, name_pattern="{}_core_web_md-3.8.0", **model_params):
```
Comment (HYLcool) [inline data_juicer/utils/model_utils.py]: We still use models in 3.7.0 version by default.

Comment (cmgzn) [reply data_juicer/utils/model_utils.py]: sorry
